### PR TITLE
[AT-341] pause_subscription

### PR DESCRIPTION
**Pull request by mshapovalov**

### CODE REVIEW

File: tests/Infrastructure/Event/Subscription/CommittingSubscriptionTest.php [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 namespace Streak\Infrastructure\Event\Subscription;
 
+use Doctrine\DBAL\Exception;
```
Comment (C3rb) [inline tests/Infrastructure/Event/Subscription/CommittingSubscriptionTest.php:16]: ?

File: tests/Infrastructure/Event/Subscription/CommittingSubscriptionTest.php [hunk 2]
```diff
@@ -544,4 +545,18 @@ public function testUnpauseWithError()
...
+            throw new Event\Exception\ListenerFailed('type', new \Exception());
+        });
+        $this->uow
+            ->expects($this->once())
+            ->method('commit')
+        ;
+        $subscription = new CommittingSubscription($this->subscription, $this->uow);
+        $this->expectException(Event\Exception\ListenerFailed::class);
+        $events = $subscription->subscribeTo($this->store);
+        iterator_to_array($events);
```
Comment (C3rb) [inline tests/Infrastructure/Event/Subscription/CommittingSubscriptionTest.php:560]: Check if it was paused

File: tests/Infrastructure/Event/Subscription/DAO/SubscriptionTest.php [hunk 3]
```diff
@@ -1445,4 +1446,46 @@ public function testCompletingListenerWhileContinuousListening()
...
+    private function createFailedListener(array $interfaces):MockObject{
+        if(1 === count($interfaces)){
+            $interfaces = $interfaces[0];
+        }
+        $result =  $this->getMockBuilder($interfaces)->getMock();
+        $result
+            ->expects($this->any())
+            ->method('on')
+            ->willReturnCallback(function () {
+                throw new Exception();
```
Comment (C3rb) [inline tests/Infrastructure/Event/Subscription/DAO/SubscriptionTest.php:1487]: It think it does not have to be `DBAL\Exception`. Maybe just use `'\Exception`

**Review by azezula — APPROVED** [review]

**Review by C3rb — APPROVED** [review]

File: tests/Infrastructure/Event/Subscription/CommittingSubscriptionTest.php (hunk 1 above)
Comment (mshapovalov) [reply tests/Infrastructure/Event/Subscription/CommittingSubscriptionTest.php]: fixed

File: tests/Infrastructure/Event/Subscription/DAO/SubscriptionTest.php (hunk 3 above)
Comment (mshapovalov) [reply tests/Infrastructure/Event/Subscription/DAO/SubscriptionTest.php]: fixed

**Pull request merged**